core/chains/evm/logpoller/log_poller.go

```diff
@@ -258,6 +259,15 @@ func (lp *logPoller) UnregisterFilter(name string, qopts ...pg.QOpt) error {
 	return nil
 }
 
+// HasFilter returns true if the log poller has an active filter with the given name.
+func (lp *logPoller) HasFilter(name string) bool {
+	lp.filterMu.RLock()
+	defer lp.filterMu.RUnlock()
+
+	_, ok := lp.filters[name]
```

Comment:
does lp load filters into memory on boot? Not sure if we need a fallback to DB here?

Reply:
Yes, it loads all filters from db on startup. I think it's fine to just check in-memory filters.

Reply:
(Unless there's a chance this would be called while LogPoller is still starting up?)

Reply:
not sure if we'll always call this function if filters are guaranteed to be loaded? Maybe we can check for filtersLoaded = true, if not load them like this 
https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/logpoller/log_poller.go#L442-L447